packages/admin_view/src/components/LexiconTable.tsx

```diff
@@ -0,0 +1,118 @@
+import { GetAllLexEntriesQuery, useGetAllLexEntriesQuery } from '../graphql/lexicon/lexicon.ts';
+import { DataGrid, GridColDef,DataGridProps } from '@mui/x-data-grid';
+import React, { useState } from 'react';
+import { Box, Stack, Button } from '@mui/material';
+import { alpha, createTheme, styled } from '@mui/material/styles';
+
+const theme = createTheme({
+  palette:{
+    primary:{
+      light: '#757ce8', 
+      main: '#3f50b5', 
+      dark: '#002884',   
+    }
+  }
+})
+const StyledDataGrid = styled(DataGrid)<DataGridProps>(({ theme }) => ({
+  '&.MuiDataGrid-root': {
+    backgroundColor: alpha(theme.palette.primary.light, 0.3),
+  },
+  //don't know why this is not applied
+  '.MuiDataGrid-columnHeaders': {
+    backgroundColor: alpha(theme.palette.primary.dark, .8),
+    fontWeight: 'bold',
+
+  },
+  '.MuiDataGrid-cell': {
+    backgroundColor: alpha(theme.palette.primary.light, 0.5),
+  },
+}));
+
+
+const tableColumns: GridColDef<GetAllLexEntriesQuery['lexiconAllEntries'][number]>[] = [
+  {
+    field: 'primary',
+    headerName: 'Primary',
+    width: 200,
+    headerAlign: 'center',
+    headerClassName: 'boldHeader'
+  },
+  {
+    field: 'associates',
+    headerName: 'Associates',
+    width: 200,
+    headerAlign: 'center',
+  },
+  {
+    field: 'fields',
+    headerName: 'Fields',
+    width: 200,
+    headerAlign: 'center',
+
+  },
+  {
+    field: 'key',
+    headerName: 'Key',
+    width: 200,
+    headerAlign: 'center',
+
+  },
+  {
+    field: 'video',
+    headerName: 'Video',
+    width: 200,
+    headerAlign: 'center',
+
+  }
+];
+
+
+const LexiconTable = () => {
```

Comment:
You are on the right track here but I think it will be much easier to adapt the crud datagrid example from the MUI docs to our needs, rather than trying to code our own solution from scratch. I would recommend starting with the example at https://mui.com/x/react-data-grid/editing/ and editing it where necessary to handle our data. I think it should be as simple as just changing the `GridColDef` and adding in the query for the lexicon data.